### PR TITLE
Skip CPU-only tests on CircleCI machines with GPU

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR makes the CircleCI machines with GPU skip the tests that don't actually need a GPU. These tests are already extensively tested in the other CPU-only workflows.

### CODE REVIEW

**Comment by NicolasHug** [comment]
This seems to be working as expected, see e.g. the [linux gpu unittest workfow](https://app.circleci.com/pipelines/github/pytorch/vision/8789/workflows/93565db9-828e-4bc5-bf32-299e53d6cde0/jobs/638175) where the `test_classification_model[cpu...` tests are skipped while the CUDA ones are run. `test_generalizedrcnn_transform_repr` is also properly skipped.

The tests aren't properly skipped on Windows because of https://github.com/pytorch/vision/issues/4006, but we can merge this anyway.

Note: There are still a lot of CPU-only tests that are run on the GPU CI. These are all the tests in tests files like `test_transforms.py` that *don't* include GPU tests at all. I will open another PR to deal with these.

File: .circleci/config.yml.in [hunk 1]
```diff
@@ -687,12 +687,20 @@ jobs:
...
+          # So instead we dump the variables we need in env.list and we pass that file when invoking "docker run".
+          name: export CIRCLECI env var
+          command: echo "CIRCLECI=true" >> ./env.list
       - run:
           name: Install torchvision
           command: docker run -t --gpus all -v $PWD:$PWD -w $PWD -e UPLOAD_CHANNEL -e CU_VERSION "${image_name}" .circleci/unittest/linux/scripts/install.sh
       - run:
           name: Run tests
-          command: docker run -e CIRCLECI -t --gpus all -v $PWD:$PWD -w $PWD "${image_name}" .circleci/unittest/linux/scripts/run_test.sh
+          command: docker run --env-file ./env.list --gpus all -v $PWD:$PWD -w $PWD "${image_name}" .circleci/unittest/linux/scripts/run_test.sh
```
Comment (NicolasHug) [inline .circleci/config.yml.in:703]: The previous `-e CIRCLECI -t` was actually enough for what we need here, but I feel like creating the file is more future proof (we might need more env variables in the future) and most importantly it allows to properly document why we need this, which can be useful  if we start using docker for more than just linux.

**Review by fmassa — APPROVED** [review]
Looks great, thanks!

With this PR we seem to have saved ~3min from the GPU CI, nice!

**Comment by datumbox** [comment]
@NicolasHug  On [unittest_windows_gpu_py3.8](https://app.circleci.com/pipelines/github/pytorch/vision/8797/workflows/afc6b9be-142d-403b-8c3d-6cf893856a24/jobs/638891), I see cuda tests being skipped. Example:
```
test/test_ops.py::TestDeformConv::test_forward[0-True-cpu] PASSED        [ 96%]
test/test_ops.py::TestDeformConv::test_forward[0-True-cuda] SKIPPED      [ 96%]
test/test_ops.py::TestDeformConv::test_forward[0-False-cpu] PASSED       [ 96%]
test/test_ops.py::TestDeformConv::test_forward[0-False-cuda] SKIPPED     [ 96%]
test/test_ops.py::TestDeformConv::test_forward[33-True-cpu] PASSED       [ 96%]
test/test_ops.py::TestDeformConv::test_forward[33-True-cuda] SKIPPED     [ 96%]
test/test_ops.py::TestDeformConv::test_forward[33-False-cpu] PASSED      [ 96%]
test/test_ops.py::TestDeformConv::test_forward[33-False-cuda] SKIPPED    [ 96%]
```

Is this related to the temporary CicleCI driver issue for win GPU?

**Pull request merged**